docs/advanced/mixin/4.shadow.md

```diff
@@ -0,0 +1,46 @@
+@Shadow
+======
+
+Shadow is a technology that redirects the access of the fields and methods from the mixin classes to the vanilla classes. It allows you to use those elements like normal ones so that your method code can be simplified. It can also help you to access the private fields or methods in the target class. 
+
+Fields
+------
+Shadowed fields are annotated with  ```@Shadow``` and have the same access modifiers, names, and types as the vanilla fields in the target class. 
+
+```java
+@Shadow
+<access modifier> <static or not> <type name> <field name>;
+```
+
+The Mixin system will automatically redirect your access to the corresponding fields in the target class.
+
+
+Methods
+-------
+Shadowed methods are annotated with  ```@Shadow``` and have the same access modifiers, names, parameter types, and return types as the vanilla methods in the target class.
+
+If the vanilla method is not static, the shadowed method should be abstract.
+
+
+```java
+@Shadow
+<access modifier> abstract <type name> <method name>(parameter list...);
+```
+
+If the vanilla method is static, the shadowed method should be static and have a method body. The return value does not matter because this method is just a placeholder.
+```java
+@Shadow
+<access modifier> static <type name> <method name>(parameter list...){
+	return null;       // for reference types
+	// return 0;       // for primitives
+	// return false;
+}
+```
+
+The Mixin system will automatically redirect your access to the corresponding methods in the target class.
```

Comment:
Not really true since it depends on other compatibilities. Edge cases should be documented.